apps/meteor/app/api/server/v1/moderation.ts

```diff
@@ -0,0 +1,240 @@
+import {
+	isReportHistoryProps,
+	isArchiveReportProps,
+	isReportInfoParams,
+	isReportMessageHistoryParams,
+	isModerationDeleteMsgHistoryParams,
+	isReportsByMsgIdParams,
+} from '@rocket.chat/rest-typings';
+import { ModerationReports, Users, Messages } from '@rocket.chat/models';
+import type { IModerationReport } from '@rocket.chat/core-typings';
+
+import { API } from '../api';
+import { deleteReportedMessages } from '../../../../server/lib/moderation/deleteReportedMessages';
+import { getPaginationItems } from '../helpers/getPaginationItems';
+
+type ReportMessage = Pick<IModerationReport, '_id' | 'message' | 'ts' | 'room'>;
+
+API.v1.addRoute(
+	'moderation.getReports',
+	{
+		authRequired: true,
+		validateParams: isReportHistoryProps,
+		permissionsRequired: ['view-moderation-console'],
+	},
+	{
+		async get() {
+			const { latest: _latest, oldest: _oldest, selector = '' } = this.queryParams;
+
+			const { count = 20, offset = 0 } = await getPaginationItems(this.queryParams);
+			const { sort } = await this.parseJsonQuery();
+
+			const latest = _latest ? new Date(_latest) : new Date();
+			const oldest = _oldest ? new Date(_oldest) : new Date(0);
+
+			const reports = await ModerationReports.findReportsGroupedByUser(latest, oldest, selector, { offset, count, sort }).toArray();
+
+			if (reports.length === 0) {
+				return API.v1.success({
+					reports,
+					count: 0,
+					offset,
+					total: 0,
+				});
+			}
+
+			const total = await ModerationReports.countReportsInRange(latest, oldest, selector);
+
+			return API.v1.success({
+				reports,
+				count: reports.length,
+				offset,
+				total,
+			});
+		},
+	},
+);
+
+API.v1.addRoute(
+	'moderation.user.getMessageHistory',
```

Comment:
no need to seperate the name. and the previous non-need of verb applies here as well. 

maybe also instead of messageHistory, we can make it more clear, like `reportedMessages`